### PR TITLE
Chapter 9 Login Advanced

### PR DESCRIPTION
**Pull request by tuanpm-0927**
Screenshot from 2019-08-05 16-24-15

### CODE REVIEW

**Comment by tuanpm-0927** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -1,17 +1,38 @@
...
-      BCrypt::Password.create(string, cost: cost)
-    end
+   # Remember login
+  def remember
+    self.remember_token = new_token
+    update_attribute(:remember_digest, digest(remember_token))
+  end
+
+  def authenticated?(remember_token)
+    return false if remember_digest.nil?
```
Comment (vanvtt-0952) [inline app/models/user.rb]: `if remember_digest.nil?` thành `unless remember_digest` dc nhé

File: test/controllers/sessions_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class SessionsControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (vanvtt-0952) [inline test/controllers/sessions_controller_test.rb]: Xóa file

**Comment by vanvtt-0952** [comment]
merged
fix comment nhỏ rồi mợt đi chú

**Pull request merged**